lib/controllers/pullController.js

```diff
@@ -27,10 +27,14 @@ module.exports = function (apiPlatformService, logger, messages,
           workspace.api.id, workspace.apiVersion.id);
       })
       .then(function (apiFilesMetadata) {
-        workspace.rootRamlPath =
-          _.find(apiFilesMetadata, 'id', workspace.apiVersion.rootFileId).path;
+        if (!_.isEmpty(apiFilesMetadata)) {
+          workspace.rootRamlPath =
+            _.find(apiFilesMetadata, 'id', workspace.apiVersion.rootFileId).path;
 
-        return workspaceRepository.update(workspace);
+          return workspaceRepository.update(workspace);
```

Comment:
if the API is empty the workspace isn't updated and then other commands (like push) will break